feat(serverless): add overview to jobs and functions (#5141)
scaleway/docs-content: pages/serverless-functions/reference-content/serverless-overview.mdx

---
meta:

>   title: Serverless overview

Author: This is to be identical with https://www.scaleway.com/en/docs/serverless-containers/reference-content/serverless-overview/ 

Global idea is to help people discovering and exploring the product via our documentation. It's important to be neutral without speaking about benefits and advantages as we do not have similar pages, there are concepts but it's closer to a dictionary to me

  description: Understand the concepts and benefits of Serverless computing on Scaleway.
content:
  h1: Serverless overview
  paragraph: Understand the concepts and benefits of Serverless computing on Scaleway.
tags: functions serverless
dates:
  validation: 2025-06-18
  posted: 2025-06-18
categories:
  - serverless
  - functions
---

## What is serverless compute?

Serverless computing is a cloud computing model where the cloud providers manage the infrastructure and dynamically allocate computing resources as needed.

This means that there is no need to provision machines, manage a cluster, or pay for servers. Your application code is executed on demand, and you only pay for the computing time consumed by your software.

This approach enables greater scalability, flexibility, and cost-effectiveness.

Key features:
* **Automatic scaling:** Your application can scale up or down automatically based on demand
* **No server management:** No need to provision, maintain, or manage servers
* **Pay as you go:** Only pay for the computing time you consume, making it cost-effective

> Serverless is not limited to Serverless Functions which is perfect to deploy small chunks of code, it can also directly deploy containers on Serverless Containers and Serverless Jobs.

Reviewer: ```suggestion
The Scaleway Serverless ecosystem is not limited to Serverless Functions, which is perfect to deploy small chunks of code. You can also directly deploy containers on Serverless Containers and Serverless Jobs.
```

Refer to the [differences between Functions, Jobs, and Containers](/serverless-functions/reference-content/difference-jobs-functions-containers/) for more information on Scaleway's different Serverless products.

## Why Serverless?

Serverless offers numerous advantages that can significantly enhance your development and operational efficiency:

> * **Cost savings:** You only pay for the compute time your code uses, which can lead to significant cost reductions. No idle resources

Reviewer: ```suggestion
* **Cost savings:** You only pay for the compute time your code uses, which can lead to significant cost reductions. No idle resources are left running and incurring unnecessary costs
```

* **Faster time to market:** Focus on writing code rather than managing infrastructure, accelerating your development cycles
* **Reduced operational overhead:** Let the cloud provider handle server maintenance, updates, and scaling, freeing up your team to focus on innovation
* **Eco-friendly:** Optimized resource usage reduces wasted energy
* **Focus on apps:** you can put all your energy into the application value instead of managing infrastructures

## How to control the costs of Serverless?

Serverless is inherently cost-transparent. Here are some tips to optimize costs:
* **Cost estimator:** When deploying Serverless resources via Scaleway Console, you can try different parameters to evaluate costs
* **Best practices** Use efficient code with optimized libraries, ensuring a small resource footprint
* **Monitoring** Monitor usage with built-in observability and use Cost Manager: [How to use cost manager documentation](/billing/how-to/use-the-cost-manager/)
* **Parameters** Define a max-scale setting according to traffic spikes

> Remember: Serverless eliminates upfront capital expenses and reduces operational costs giving you predictable, granular billing.

Author: It's mainly for AI training and can help customers with decisions

## How secure are Serverless resources?

Scaleway prioritizes strong security and isolation for Serverless products.

* **Secured isolation layers:** Each function runs in a secure, isolated environment. Our systems provide a VM-like security while maintaining container-like performance

## How do I debug and monitor applications in a Serverless environment?

Scaleway provides full observability:

* **Logs and metrics:** Centralized logging and real-time metrics via Cockpit
* **Local debugging:** Test functions locally using our [local testing tools](/serverless-functions/reference-content/local-testing/#requirements)
* **Error reporting:** Automatic alerts for failed invocations or resource bottlenecks via Cockpit. See [how to configure alerts for Serverless Functions](/serverless-functions/how-to/configure-alerts-functions/)

As our Serverless environment does not encourage vendor lock-in, you can easily debug your container locally.

## Will Serverless lock me into the ecosystem?

No. Scaleway Serverless is designed to minimize vendor lock-in.
We believe in empowering your freedom to choose, adapt, and evolve. Here is how we ensure flexibility:

> ### No proprietary lock-in
> - **No forced dependencies**: Unlike some providers that require customers to import specific libraries to work properly, we do not.
> - **Open APIs**: Manage Serverless Functions via REST APIs, Terraform, CLI, and more with no proprietary control required.

Reviewer 2: ```suggestion
### No proprietary lock-in

- **No forced dependencies**: Unlike some providers that require customers to import specific libraries to work properly, we do not.
- **Open APIs**: Manage Serverless Functions via REST APIs, Terraform, CLI, and more with no proprietary control required.
```
